Neos.Fusion/Classes/FusionObjects/DataStructureImplementation.php

```diff
@@ -29,73 +23,18 @@ class DataStructureImplementation extends AbstractArrayFusionObject
      */
     public function evaluate()
     {
-        $sortedChildFusionKeys = $this->sortNestedFusionKeys();
-
-        if (count($sortedChildFusionKeys) === 0) {
-            return [];
-        }
-
-        $result = [];
-        foreach ($sortedChildFusionKeys as $key) {
-            $propertyPath = $key;
-            if ($this->isUntypedProperty($this->properties[$key])) {
-                $propertyPath .= '<Neos.Fusion:DataStructure>';
-            }
-            try {
-                $value = $this->fusionValue($propertyPath);
-            } catch (\Exception $e) {
-                $value = $this->runtime->handleRenderingException($this->path . '/' . $key, $e);
-            }
-            if ($value === null && $this->runtime->getLastEvaluationStatus() === Runtime::EVALUATION_SKIPPED) {
-                continue;
-            }
-            $result[$key] = $value;
-        }
-
-        return $result;
-    }
-
-    /**
-     * Sort the Fusion objects inside $this->properties depending on:
-     * - numerical ordering
-     * - position meta-property
-     *
-     * This will ignore all properties defined in "@ignoreProperties" in Fusion
-     *
-     * @see PositionalArraySorter
-     *
-     * @return array an ordered list of key value pairs
-     * @throws Fusion\Exception if the positional string has an unsupported format
-     */
-    protected function sortNestedFusionKeys()
-    {
-        $arraySorter = new PositionalArraySorter($this->properties, '__meta.position');
-        try {
-            $sortedFusionKeys = $arraySorter->getSortedKeys();
-        } catch (InvalidPositionException $exception) {
-            throw new Fusion\Exception('Invalid position string', 1345126502, $exception);
-        }
-
-        foreach ($this->ignoreProperties as $ignoredPropertyName) {
-            $key = array_search($ignoredPropertyName, $sortedFusionKeys);
-            if ($key !== false) {
-                unset($sortedFusionKeys[$key]);
-            }
-        }
-        return $sortedFusionKeys;
+        return $this->evaluateNestedProperties('Neos.Fusion:DataStructure');
     }
 
     /**
      * Returns TRUE if the given property has no object type assigned
```

Comment:
This docblock needs to be adjusted. And the return type set I assume?

Reply:
Is adjusted